diff --git a/h5peditor.class.php b/h5peditor.class.php
index c1a3fb5..37a60a5 100644
--- a/h5peditor.class.php
+++ b/h5peditor.class.php
@@ -373,7 +373,7 @@ class H5peditor {
    *
    * @return array Libraries that was requested
    */
-  public function getLibraryData($machineName, $majorVersion, $minorVersion, $languageCode, $prefix = '', $fileDir = '', $defaultLanguage) {
+  public function getLibraryData($machineName, $majorVersion, $minorVersion, $languageCode, $prefix = '', $fileDir = '', $defaultLanguage = '') {
     $libraryData = new stdClass();
 
     $library = $this->h5p->loadLibrary($machineName, $majorVersion, $minorVersion);
